distribution/lib/Standard/Base/0.0.0-dev/src/Random.enso

```diff
@@ -1,99 +1,307 @@
+       Return a random `Float` in the specified range.
+
+       Arguments:
+       - min: The minimum value of the range to pick from (inclusive).
+       - max: The maximum value of the range to pick from (exclusive).
+
+       > Example
+         Generate a float between 2.3 and 3.4.
+
+              import Standard.Base.Random.Random
+
+              i = Random.float 2.3 3.4
+    float : Float -> Float -> Float
+    float (min:Float=0.0) (max:Float=1.0) = Random.default_instance.float min max
 
     ## GROUP Random
-       Gets the next random Float from a normal distribution with mean 0 and std-dev 1.
-    gaussian : Float
-    gaussian self =
-        self.java_random.nextGaussian
+       Return a random `Boolean` using the default instance of
+       `Random_Generator`.
+
+       > Example
+         Generate a random boolean.
+
+             import Standard.Base.Random.Random
+
+             b = Random.boolean
+    boolean : Boolean
+    boolean = Random.default_instance.boolean
 
     ## GROUP Random
-       Gets the next random integer between min (inclusive) and max (exclusive).
+       Return a random Gaussian-distributed `Float` with the specified mean and
+       standard deviation.
 
        Arguments:
-        - min: the minimum value (inclusive) of the random integer.
-        - max: the maximum value (exclusive) of the random integer.
-    integer : Integer -> Integer -> Integer
-    integer self min=0 max=100 =
-        range = max - min
-        if range < Java_Integer.MAX_VALUE then min + (self.java_random.nextInt range) else
-            Error.throw (Illegal_Argument.Error "Currently only integer ranges of up to 2^31-1 are supported.")
+       - standard_deviation: The standard deviation of the sampled distribution.
+       - mean: The mean of the sampled distribution.
+
+       > Example
+         Generate an gaussian float.
+
+              import Standard.Base.Random.Random
+
+              f = Random.gaussian
+    gaussian : Float -> Float -> Float
+    gaussian (standard_deviation:Float=1.0) (mean:Float=0.0) = Random.default_instance.gaussian standard_deviation mean
+
+    ## GROUP Random
+       Returns `count` indices sampled from the range 0 (inclusive) to n
+       (exclusive) without replacement.
+       
+       Arguments:
+       - n: The end of the range to sample from (exclusive).
+       - count: The number of samples to take.
+
+       > Example
+         Generate 10 random indices between 0 and 100.
+
+              import Standard.Base.Random.Random
+
+              indices = Random.indices 100 10
+    indices : Integer -> Integer -> Vector Integer
+    indices (n:Integer) (count:Integer=1) = Random.default_instance.indices n count
+
+    ## GROUP Random
+       Return a `Date` within the specified `Date_Range`.
+
+       Arguments:
+       - date_range: The `Date_Range` that specifies the range to pick from. The
+         range is inclusve at the lower bound, and exclusive at the upper bound.
```

Comment:
Note that we can actually make the end inclusive within `up_to`, and I think the example should reflect that:
```
d = Random.date (my_first_day_of_vacation.up_to my_last_day_of_vacation include_end=True)
```

It seems more intuitive to me to use this method with inclusive end - i.e. when deciding on which day of my vacation to go hiking - I'd put the first and last day as the range - not the '1 day after last'.

And there are reasons why `Date_Range` is right-exclusive by default I guess, but I think in the example we could show the right-inclusive example to show how to most easily use this.

Reply:
I'll have an example for both, since it's a common source of errors. Removed the note about it being exclusive at the end, since that's about the `Range` class, not this class, and the examples make it clear.